packages/dds/tree/src/test/simple-tree/api/schemaCreationUtilities.spec.ts

```diff
@@ -235,4 +235,99 @@ describe("schemaCreationUtilities", () => {
 			const _b: InsertableImplicit = new DayNodes.Tomorrow();
 		}
 	});
+
+	// Highest priority issue
+	it("enum interop - adaptEnum with explicit schema", () => {
+		enum Day {
+			Today = "today",
+			Tomorrow = "tomorrow",
+		}
+
+		function getDay(value: string): Day {
+			switch (value) {
+				case "today":
+					return Day.Today;
+				case "tomorrow":
+					return Day.Tomorrow;
+				default:
+					throw new Error(`Invalid value: ${value}`);
+			}
+		}
+
+		const DayNodes = adaptEnum(schema, Day);
+
+		class DayObject extends schema.object("DayObject", {
+			value: DayNodes.schema,
+			name: schema.string,
+		}) {}
+		function createDayObjects(value: Day, name: string): DayObject[] {
+			// @ts-expect-error Day is not compatible with value of Day.Today
+			const dayObject1 = new DayObject({ value: DayNodes("today"), name: "1" });
+			// @ts-expect-error Day is not compatible with value of Day.Today
+			const dayObject4 = new DayObject({ value: getDay("today"), name: "4" });
```

Comment:
you need to provide an actual instance of your node type here, not an enum value.

The suggestion below is intended to work, but there seems to be a bug which prevents it from building:
```suggestion
			const dayObject4 = new DayObject({ value: DayNodes(getDay("today")), name: "4" });
```

It seems that unions no longer correctly work when provided to enum schema to construct from.
I suspect this is a recent regression and I'll look for a fix.

As a workaround you can do the rather stupid seeming:

```suggestion
			const dayObject4 = new DayObject({ value: DayNodes(getDay("today") as Day.Today), name: "4" });
```

to get it to compile if you need to get unblocked.

Reply:
I have a fix for the bug. A PR will be posted shortly. Thanks for having a nice repro with a test: it helped track it down a lot!